src/fsharp/InfoReader.fs

```diff
@@ -107,7 +108,7 @@ type PropertyCollector(g, amap, m, ty, optFilter, ad) =
                 PropInfosEquivByNameAndPartialSig EraseNone g amap m pinfo1 pinfo2 &&
                 pinfo1.IsDefiniteFSharpOverride = pinfo2.IsDefiniteFSharpOverride )
 
-    let props = new Dictionary<PropInfo, PropInfo>(hashIdentity)
+    let props = new ConcurrentDictionary<PropInfo, PropInfo>(hashIdentity)
```

Comment:
This doesn't have to be concurrent, it's just a transient helper to collect up getter/setter pairs

Reply:
We can leave this in since the ordering somehow fixes (or changes) the property ordering and you've already adjusted the error messages